Source/WebCore/platform/graphics/skia/SkiaUnacceleratedThreadedRenderingPool.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "SkiaUnacceleratedThreadedRenderingPool.h"
+
+#if USE(COORDINATED_GRAPHICS) && USE(SKIA)
+#include "DisplayListReplayer.h"
+#include "GraphicsContextSkia.h"
+#include "NicosiaBuffer.h"
+
+namespace WebCore {
+
+SkiaUnacceleratedThreadedRenderingPool::SkiaUnacceleratedThreadedRenderingPool(unsigned numCpuPaintingThreads)
+    : m_workerPool(WorkerPool::create("SkiaPaintingThread"_s, numCpuPaintingThreads))
+{
+}
+
+void SkiaUnacceleratedThreadedRenderingPool::postPaintingTask(const IntRect& initialClip, const Ref<Nicosia::Buffer>& buffer, std::unique_ptr<DisplayList::DisplayList>&& displayList)
+{
+    m_workerPool->postTask([initialClip, buffer = Ref { buffer } , displayList = WTFMove(displayList)] {
+        RELEASE_ASSERT(buffer->surface());
+        GraphicsContextSkia context(sk_ref_sp(buffer->surface()));
+
+        DisplayList::Replayer replayer(context, *displayList.get());
+        replayer.replay(initialClip);
+
+        buffer->completePainting();
+    });
```

Comment:
So, in the end this class is just wrapping a WorkerPool, and this postPaintingTask makes its contents separated from the recording part, where buffer->beginPainting happens. I think we can just create the WorkerPool in CompositingCoorindator and add a getter and simply call postTask instead of this wrapper.

Reply:
Done.